<h3>1.- Importación de librerías</h3>

In [7]:
# Importamos las librerías necesarias

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

<h3>2.- Importación del Dataset</h3>

In [8]:
# Importamos el Dataset leyendo el archivo csv en un DataFrame

df_bmw0 = pd.read_csv("dataset/bmw_pricing_v3.csv")

<h3>3.- Exploración Inicial del Dataset</h3>

Ejemplo de columnas

In [9]:
df_bmw0.sample(10)

marca modelo        km  potencia fecha_registro tipo_gasolina  color  \
1534   NaN    318  124831.0     100.0     2013-08-01        diesel   grey   
567    BMW    325   77499.0     160.0            NaN        diesel  white   
1326   BMW    330  178928.0     180.0            NaN        diesel  black   
4356   BMW     X3   79685.0     190.0     2014-02-01        diesel  black   
130    NaN    325  221500.0     155.0     2009-07-01        petrol   grey   
642    BMW    318  172160.0     100.0     2013-02-01        diesel  black   
1064   BMW    318  125372.0     105.0     2014-03-01        diesel   blue   
4000   BMW     X1  114569.0     105.0     2014-01-01        diesel    NaN   
1865   BMW    116   47793.0      85.0            NaN        diesel  black   
834    BMW    318  148429.0     100.0     2013-06-01        diesel    NaN   

     tipo_coche volante_regulable aire_acondicionado camara_trasera  \
1534        NaN              True               True          False   
567      estate             False              False          False   
1326     estate              True               True          False   
4356        suv             False              False          False   
130       coupe              True               True           True   
642      estate              True               True          False   
1064        NaN             False                NaN          False   
4000        suv             False               True          False   
1865  hatchback             False               True           True   
834      estate             False               True          False   

     asientos_traseros_plegables elevalunas_electrico bluetooth    gps  \
1534                         NaN                False     False   True   
567                        False                False     False   True   
1326                       False                 True      True   True   
4356                         NaN                False     False  False   
130                        False                 True       NaN   True   
642                        False                False     False   True   
1064                         NaN                False     False   True   
4000                         NaN                 True       NaN   True   
1865                         NaN                False      True   True   
834                        False                False      True   True   

     alerta_lim_velocidad   precio fecha_venta  
1534                  NaN  10500.0  2018-07-01  
567                 False  14900.0  2018-03-01  
1326                 True  11300.0  2018-06-01  
4356                  NaN    100.0  2018-05-01  
130                   NaN   9300.0  2018-07-01  
642                 False   9500.0  2018-03-01  
1064                 True  13400.0  2018-05-01  
4000                  NaN  14800.0  2018-03-01  
1865                False  15100.0  2018-02-01  
834                 False  10500.0  2018-04-01

Número de filas y columnas

In [10]:
df_bmw0.shape

(4843, 18)

Columnas y tipo de Datos

In [11]:
df_bmw0.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4843 entries, 0 to 4842
Data columns (total 18 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   marca                        3873 non-null   object 
 1   modelo                       4840 non-null   object 
 2   km                           4841 non-null   float64
 3   potencia                     4842 non-null   float64
 4   fecha_registro               2420 non-null   object 
 5   tipo_gasolina                4838 non-null   object 
 6   color                        4398 non-null   object 
 7   tipo_coche                   3383 non-null   object 
 8   volante_regulable            4839 non-null   object 
 9   aire_acondicionado           4357 non-null   object 
 10  camara_trasera               4841 non-null   object 
 11  asientos_traseros_plegables  1452 non-null   object 
 12  elevalunas_electrico         4841 non-null   object 
 13  bluetooth         

Número de nulos

In [12]:
df_bmw0.isnull().sum()

marca                           970
modelo                            3
km                                2
potencia                          1
fecha_registro                 2423
tipo_gasolina                     5
color                           445
tipo_coche                     1460
volante_regulable                 4
aire_acondicionado              486
camara_trasera                    2
asientos_traseros_plegables    3391
elevalunas_electrico              2
bluetooth                       728
gps                               0
alerta_lim_velocidad            728
precio                            6
fecha_venta                       1
dtype: int64

# Comentarios

 - El Dataset tiene 4843 registros

 - Todas las columnas (excepto la de gps) tienen algún nulo, aunque hay 9 de ellas que tienen solo algunos valores nulos sueltos que podrán imputarse fácilmente.

 - Hay 3 columnas (color, bluetooth y alerta_lim_velocidad) que tienen un número moderado de nulos (entre el 10% y el 20% del total de registros)

 - La columna tipo_coche tiene un 33% de valores nulos

 - La columna fecha_registro tiene un 50% de valores nulos

 - La columna asientos_traseros_plegables tiene un 70% de valores nulos

 - Tenemos 3 columnas numéricas (km, potencia y precio), 1 columna booleana (gps), y el resto de las columnas son objects (strings)

 - A la vista de los datos, fecha_registro y fecha_venta podrían transformarse a tipo Date

 - A la vista de los datos, las columnas volante_regulable, aire-acondicionado, camara_trasera, asientos_traseros_plegables, 
elevalunas_electrico, bluetooh, alerta_lim_velocidad podrían transformarse a tipo Booleano.